java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -0,0 +1,1702 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/aggregation.hpp>
+#include <cudf/binaryop.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/concatenate.hpp>
+#include <cudf/lists/extract.hpp>
+#include <cudf/reshape.hpp>
+#include <cudf/lists/detail/concatenate.hpp>
+#include <cudf/datetime.hpp>
+#include <cudf/filling.hpp>
+#include <cudf/hashing.hpp>
+#include <cudf/quantiles.hpp>
+#include <cudf/reduction.hpp>
+#include <cudf/replace.hpp>
+#include <cudf/rolling.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/search.hpp>
+#include <cudf/strings/attributes.hpp>
+#include <cudf/strings/capitalize.hpp>
+#include <cudf/strings/case.hpp>
+#include <cudf/strings/char_types/char_types.hpp>
+#include <cudf/strings/combine.hpp>
+#include <cudf/strings/contains.hpp>
+#include <cudf/strings/convert/convert_booleans.hpp>
+#include <cudf/strings/convert/convert_datetime.hpp>
+#include <cudf/strings/convert/convert_floats.hpp>
+#include <cudf/strings/convert/convert_integers.hpp>
+#include <cudf/strings/extract.hpp>
+#include <cudf/strings/find.hpp>
+#include <cudf/strings/padding.hpp>
+#include <cudf/strings/replace.hpp>
+#include <cudf/strings/replace_re.hpp>
+#include <cudf/strings/split/split.hpp>
+#include <cudf/strings/strip.hpp>
+#include <cudf/strings/substring.hpp>
+#include <cudf/transform.hpp>
+#include <cudf/unary.hpp>
+#include <cudf/utilities/bit.hpp>
+#include <cudf/lists/lists_column_view.hpp>
+#include <cudf/structs/structs_column_view.hpp>
+#include <map_lookup.hpp>
+
+#include "cudf_jni_apis.hpp"
+#include "dtype_utils.hpp"
+
+
+extern "C" {
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_upperStrings(JNIEnv *env, jobject j_object,
+                                                                      jlong handle) {
+  JNI_NULL_CHECK(env, handle, "column is null", 0);
+
+  try {
+    cudf::jni::auto_set_device(env);
+    cudf::column_view *column = reinterpret_cast<cudf::column_view *>(handle);
+    cudf::strings_column_view strings_column(*column);
+
+    std::unique_ptr<cudf::column> result = cudf::strings::to_upper(strings_column);
+    return reinterpret_cast<jlong>(result.release());
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_lowerStrings(JNIEnv *env, jobject j_object,
+                                                                      jlong handle) {
```

Comment:
Nit: Indentation is off on line-wrapped parameters for many functions in this file.